src/violations.rs

```diff
@@ -768,44 +811,30 @@ impl Violation for AssertTuple {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub enum IsCmpop {
-    Is,
-    IsNot,
-}
-
-impl From<&Cmpop> for IsCmpop {
-    fn from(cmpop: &Cmpop) -> Self {
-        match cmpop {
-            Cmpop::Is => IsCmpop::Is,
-            Cmpop::IsNot => IsCmpop::IsNot,
-            _ => unreachable!("Expected Cmpop::Is | Cmpop::IsNot"),
-        }
-    }
-}
-
 define_violation!(
-    pub struct IsLiteral(pub IsCmpop);
+    pub struct IsLiteral(pub ViolationsCmpop);
 );
 impl AlwaysAutofixableViolation for IsLiteral {
     fn message(&self) -> String {
         let IsLiteral(cmpop) = self;
         match cmpop {
-            IsCmpop::Is => "Use `==` to compare constant literals".to_string(),
-            IsCmpop::IsNot => "Use `!=` to compare constant literals".to_string(),
+            ViolationsCmpop::Is => "Use `==` to compare constant literals".to_string(),
+            ViolationsCmpop::IsNot => "Use `!=` to compare constant literals".to_string(),
+            _ => unreachable!(),
```

Comment:
I think you should rather implement a 3rd wrapper. I think merging enums like this is a bad idea because it looses compile-time safety, as evidenced by the `unreachable!()` statements that you had to add. Runtime panics are bad and it's better if the compiler can check that the match blocks cover all cases.

Furthermore we want to split up this file in the future in the future (moving the struct definitions to the respective rule files) which also is easier if we don't artificially merge unrelated enums.

Reply:
Updated to be a 3rd wrapper, I was coming at the problem from the point of view of side stepping the `Serialize + Deserialize + Eq` requirements that are not found on `Cmpop`. However, as you point out the wrappers are also used for type safety in other cases.